### PR TITLE
refactor(query): avoid copy in io worker

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

refactor(query): avoid copy in io worker

Closes #issue

### CODE REVIEW

File: src/query/storages/fuse/src/io/read/block_reader.rs [hunk 1]
```diff
@@ -33,7 +40,11 @@ use common_storage::ColumnLeaf;
 use common_storage::ColumnLeaves;
 use common_storages_table_meta::meta::ColumnMeta;
 use futures::future::try_join_all;
+use futures::Stream;
+use futures_util::TryStreamExt;
+use opendal::raw::output::Read as DALRead;
```
Comment (Xuanwo) [inline src/query/storages/fuse/src/io/read/block_reader.rs:45]: Can we use `futures::Stream<Item=io::Result<Bytes>>` directly? It's better not to depend on opendal's `raw` API.

File: src/query/storages/fuse/src/io/read/block_reader.rs [hunk 2]
```diff
@@ -339,9 +440,26 @@ impl BlockReader {
...
+                cx: &mut Context<'_>,
+            ) -> Poll<Option<Self::Item>> {
+                self.reader.poll_next(cx)
+            }
+        }
+
+        let stream = ReaderStream {
+            reader: o.range_reader(start..end).await?,
+        };
+        Ok((index, stream.try_collect::<Vec<_>>().await?))
```
Comment (Xuanwo) [inline src/query/storages/fuse/src/io/read/block_reader.rs:462]: The following code should work:

```shell
o.range_reader(start..end).await?.try_collect::<Vec<_>>().await?
```

**Comment by BohuTANG** [comment]
I will bench this PR later

**Comment by BohuTANG** [comment]
@mergify update

File: src/query/storages/fuse/src/io/read/block_reader.rs [hunk 3]
```diff
@@ -71,6 +119,57 @@ impl OwnerMemory {
...
+}
+
+impl<'a> Read for ChunksReader<'a> {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        while let Some(first) = self.bytes.front_mut() {
+            match first.read(buf)? {
+                0 if !buf.is_empty() => {
+                    self.bytes.pop_front();
+                }
+                n => return Ok(n),
```
Comment (Xuanwo) [inline src/query/storages/fuse/src/io/read/block_reader.rs:152]: buf could be much larger than single chunk of bytes. We need to check `buf.is_empty` and kept writing into it.

**Comment by Xuanwo** [comment]
Hello, due to the high volume of PRs that need to be tracked in Databend, I am closing this inactive PR for now. You are welcome to reopen it if you wish to continue working on it.

**Pull request closed without merging**